pkg/helm/urlfetcher.go

```diff
@@ -15,83 +15,89 @@
 package helm
 
 import (
-	"bytes"
 	"crypto/sha256"
 	"encoding/hex"
 	"fmt"
 	"io"
 	"io/ioutil"
-	"net/http"
 	"net/url"
 	"os"
+	"path"
 	"path/filepath"
 	"strings"
 
 	"github.com/mholt/archiver"
 
+	"istio.io/operator/pkg/httprequest"
 	"istio.io/operator/pkg/util"
 	"istio.io/pkg/log"
 )
 
-// FileDownloader is wrapper of HTTP client to download files
-type FileDownloader struct {
-	// client is a HTTP/HTTPS client.
-	client *http.Client
-}
+const (
+	// InstallationPathTemplate is used to construct installation url based on version
+	InstallationPathTemplate = "https://github.com/istio/istio/releases/download/%s/istio-%s-linux.tar.gz"
+	// ChartsTempFilePrefix is temporary Files prefix
+	ChartsTempFilePrefix = "istio-install-package"
+	// ChartsFilePath is file path of installation packages to helm charts.
+	ChartsFilePath = "install/kubernetes/operator/charts"
+	// SHAFileSuffix is the default SHA file suffix
+	SHAFileSuffix = ".sha256"
+)
 
 // URLFetcher is used to fetch and manipulate charts from remote url
 type URLFetcher struct {
 	// url is url to download the charts
 	url string
 	// verifyURL is url to download the verification file
 	verifyURL string
-	// versionsURL is url to download version file
-	versionsURL string
 	// verify indicates whether the downloaded tar should be verified
 	verify bool
 	// destDir is path of charts downloaded to, empty as default to temp dir
 	destDir string
-	// downloader downloads files from remote url
-	downloader *FileDownloader
 }
 
-// NewURLFetcher creates an URLFetcher pointing to urls and destination
-func NewURLFetcher(repoURL string, destDir string, chartName string, shaName string) *URLFetcher {
+// NewURLFetcher creates an URLFetcher pointing to installation package url and destination
+func NewURLFetcher(insPackageURL string, destDir string) (*URLFetcher, error) {
 	if destDir == "" {
 		destDir = filepath.Join(os.TempDir(), ChartsTempFilePrefix)
 	}
+	if _, err := os.Stat(destDir); os.IsNotExist(err) {
+		err := os.Mkdir(destDir, os.ModeDir|os.ModePerm)
+		if err != nil {
+			return nil, err
+		}
+	}
 	uf := &URLFetcher{
-		url:         repoURL + "/" + chartName,
-		verifyURL:   repoURL + "/" + shaName,
-		versionsURL: repoURL + "/" + VersionsFileName,
-		verify:      true,
-		destDir:     destDir,
-		downloader:  NewFileDownloader(),
+		url:       insPackageURL,
+		verifyURL: insPackageURL + SHAFileSuffix,
+		verify:    true,
+		destDir:   destDir,
 	}
-	return uf
+	return uf, nil
+}
+
+// GetDestDir returns path of destination dir.
+func (f *URLFetcher) GetDestDir() string {
```

Comment:
usual Go idiom is to say DestDir, not GetDestDir if possible.

Reply:
done